### PR TITLE
Register-PSSessionConfiguration fails if SesionConfig folder doesn't exist

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
Create SessionConfig folder if it doesn't exist for Register-PSSessionConfiguration

### CODE REVIEW

**Review by PaulHigin — DISMISSED** [review]
LGTM

**Comment by SteveL-MSFT** [comment]
@PaulHigin Windows PowerShell 5.1 needs this change as well before this test will pass, right (since the default remoting endpoint still goes to Windows PowerShell)?  If so, I'll change this test to `Pending`

**Comment by PaulHigin** [comment]
@SteveL-MSFT I thought that we use a new plugin for PowerShell 6.0.  Not sure but I know a new plugin was created for Core.  @JamesWTruher can you comment?

Regarding PowerShell 5.1, the SessionConfig folder should always be created via Windows manifest.  I feel this change is only needed for the case where the folder is inadvertently deleted (through some previous test) or for PowerShell 6.0 when it is never created.

**Comment by SteveL-MSFT** [comment]
Sync'd with @PaulHigin off line, this test will still fail because the default endpoint still points to 5.1 which doesn't have the fix to expose `-PowerShellVersion` yet.  Rather than making this test pending, I think it makes sense to have the default remoting to go to 6.0, I'll look into that.

**Comment by daxian-dbw** [comment]
> Rather than making this test pending, I think it makes sense to have the default remoting to go to 6.0, I'll look into that.

@PaulHigin @SteveL-MSFT Will that be a separate PR? If so, I will merge this one.

**Comment by SteveL-MSFT** [comment]
@daxian-dbw I'm going to make it part of this PR otherwise the test will still fail

**Comment by SteveL-MSFT** [comment]
I'm going to make this test `Pending` as it's blocked by https://github.com/PowerShell/PowerShell/issues/4275

**Review by daxian-dbw — APPROVED** [review]

**Comment by daxian-dbw** [comment]
@SteveL-MSFT one question before merging the PR: the test in question has been there for a long time, so why did it start to fail recently?

**Comment by SteveL-MSFT** [comment]
@daxian-dbw that specific test was added recently by @PaulHigin https://github.com/SteveL-MSFT/PowerShell/commit/7fa0dd054992082787bd1ef7e99bc0f2c9e2a05c

**Comment by PaulHigin** [comment]
It looks like similar implicit remoting tests in Implicit.Remoting.Tests.ps1 are disabled for core and we should do the same for this test.  Is there an Issue for this?

File: src/System.Management.Automation/engine/remoting/commands/CustomShellCommands.cs
Comment (iSazonov) [inline src/System.Management.Automation/engine/remoting/commands/CustomShellCommands.cs]: Sorry for being late - what is the user haven't permissions?

Comment (daxian-dbw) [inline src/System.Management.Automation/engine/remoting/commands/CustomShellCommands.cs]: Good catch! For example, on Linux, powershell is installed at /opt/microsoft/powershell, and you don't have permission to create a folder unless using `sudo`.

Comment (PaulHigin) [inline src/System.Management.Automation/engine/remoting/commands/CustomShellCommands.cs]: @iSazonov Thanks, we will have to think about how this will work and where configuration information goes for Linux cases, but for now I think letting an "access denied" error propagate is fine since these cmdlets cannot work correctly without access.

Comment (PaulHigin) [inline src/System.Management.Automation/engine/remoting/commands/CustomShellCommands.cs]: I should mention that these cmdlets are currently implemented only on Windows platform.

File: test/powershell/Modules/Microsoft.PowerShell.Utility/Implicit.Remoting.Tests.ps1
Comment (PaulHigin) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/Implicit.Remoting.Tests.ps1]: I just noticed that similar implicit remoting tests are disabled using special checks:
```
# Skip tests for CoreCLR for now
# Skip tests on ARM
$skipTest = $IsCoreCLR -or $env:PROCESSOR_ARCHITECTURE -eq 'ARM'

if ($skipTest) { return }
```
We should make this test be the same since remoting really does not work yet with PowerShell 6.0 based endpoints.  Or maybe we should mark all tests that aren't supported as "pending" since we no longer support FullCLR.

Comment (SteveL-MSFT) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/Implicit.Remoting.Tests.ps1]: @PaulHigin can you open a separate issue for this?

Comment (PaulHigin) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/Implicit.Remoting.Tests.ps1]: Done.  Issue #4289

File: src/System.Management.Automation/engine/remoting/commands/CustomShellCommands.cs
Comment (iSazonov) [inline src/System.Management.Automation/engine/remoting/commands/CustomShellCommands.cs]: Could you please open a tracking Issue?

**Pull request merged**
